### PR TITLE
Close previous stream before opening next one when compacting files

### PR DESCRIPTION
**Pull request by bradyoo**
Previous implementation of CompactDefinition opened an arbitrary number of input files to be fed through the SequentialCompactor. This is very wasteful because only one input file needs to be open at a time to compact files sequentially.

### CODE REVIEW

File: job/server/src/main/java/alluxio/job/plan/transform/CompactDefinition.java [hunk 1]
```diff
@@ -174,38 +172,29 @@ private boolean shouldIgnore(URIStatus status) {
...
-          context.getFileSystem().delete(output); // output is the compacted file
-        } catch (Throwable e) {
-          t.addSuppressed(e);
-        }
-        closer.rethrow(t);
       } finally {
-        if (!closed) {
-          closer.close();
-        }
+        context.getFileSystem().delete(output); // output is the compacted file
```
Comment (gpang) [inline job/server/src/main/java/alluxio/job/plan/transform/CompactDefinition.java]: the output will be deleted all the time?

Comment (bradyoo) [reply job/server/src/main/java/alluxio/job/plan/transform/CompactDefinition.java]: Oops. Yeah that's wrong :/

Comment (bradyoo) [reply job/server/src/main/java/alluxio/job/plan/transform/CompactDefinition.java]: actually, let me just remove this line.

File: job/server/src/main/java/alluxio/job/plan/transform/CompactDefinition.java [hunk 2]
```diff
@@ -174,37 +172,26 @@ private boolean shouldIgnore(URIStatus status) {
...
-              config.getPartitionInfo())));
-        }
-        TableSchema schema = readers.get(0).getSchema();
-        writer = closer.register(TableWriter.create(schema, output));
-        compactor.compact(readers, writer);
-      } catch (Throwable t) {
-        closer.close();
-        closed = true;
-        try {
-          context.getFileSystem().delete(output); // output is the compacted file
```
Comment (gpang) [inline job/server/src/main/java/alluxio/job/plan/transform/CompactDefinition.java]: Shouldn't we delete the output if there is a failure?

Comment (bradyoo) [reply job/server/src/main/java/alluxio/job/plan/transform/CompactDefinition.java]: Sorry. I just needed something quick to retrigger build. I went back to deleting it again.

**Comment by bradyoo** [comment]
test was successful

**Comment by bradyoo** [comment]
Tests passed and testing on autobots went well.

**Comment by gpang** [comment]
alluxio-bot, merge this please

**Comment by gpang** [comment]
alluxio-bot, cherry-pick this to branch-2.2 please

**Pull request merged**